himbaechel/uarch/gowin/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+message(STATUS "Configuring Himbaechel-Gowin uarch")
+cmake_minimum_required(VERSION 3.5)
+project(himbaechel-gowin-chipdb NONE)
+
+find_package(Python3 3.5 REQUIRED COMPONENTS Interpreter)
+set(ALL_HIMBAECHEL_GOWIN_DEVICES GW1N-1 GW1NZ-1 GW1NS-2 GW1N-4 GW1N-9 GW1N-9C GW1NS-4 GW2A-18 GW2A-18C)
+set(HIMBAECHEL_GOWIN_DEVICES ${ALL_HIMBAECHEL_GOWIN_DEVICES} CACHE STRING
```

Comment:
can we have this default to none for now so building nextpnr himbaechel doesn't require the gowin binary?

I still want to work a bit more on the user experience for building nextpnr-himbaechel and device databases, then maybe we can revisit this when nextpnr-himbaechel has replaced the old nextpnr-gowin.

Reply:
Could you tell me more about the build process of nextpnr-himbaechel and device databases? This is something I care about in YoWASP. It is not feasible to ship one binary that includes every binary and every chipdb due to size constraints (about 100 MB) and I would rather avoid configuring and building nextpnr several times if at all feasible.

Reply:
The idea is to provide a unified framework for arches to share one nextpnr binary with a reduced amount of per-arch code, and with a relatively compact deduplicated representation. Gowin will move there first and I will also look at xo2/3 moving to it. Longer term if it works out there will be more migrations and new arches using it.

It is going to have the chipdb binaries as data files (which should in every case be significantly less than 100MB). These will be configurable in the build process and in the next few days there should be a nice discovery process for those so you can pass the name of a device on the command line and it will load the right subarch and chipdb automatically (it'll be a relative path based approach similar to Yosys, I need to look exactly what we do for different arches and build environments there still).

I would guess, once it reaches a point of being end user ready (i.e. the legacy nextpnr-gowin is formally deprecated and then removed in favour of this), we'd have one nextpnr-himbaechel package and then another containing the chip databases per arch.

Reply:
Got it. Can you make sure every architecture (~ user-visible package) gets its own directory under `share/` and nothing outside of this directory is instaled? Then I will be able to ship this nicely in YoWASP.

Reply:
Yeah, that matches what I was planning

Reply:
Fantastic! Then what I will do in `yowasp-nextpnr-himbaechel` is to pull in all of the chipdbs that `yowasp-nextpnr-${ARCH}` have registered using the standard Python "entry points" (plugin) mechanism and mount them under `share/`. Ideally that in itself will be enough.

Do you plan to have e.g. a `nextpnr-ecp5` runner that proxies to `nextpnr-himbaechel --uarch ecp5`? This could be done by snooping `argv[0]` (checking if it ends with `-ecp5`, etc) and it would greatly simplify migration for Amaranth.

Reply:
Yes, this could definitely be done as part of the migration path. Although I realise a bit more work around the himbaechel startup and API is needed to support arbitrary command line arguments, which is needed both for backwards compatibility and a friendly CLI in general.